src/Access/LDAPAccessStorage.cpp

```diff
@@ -320,6 +319,10 @@ std::set<String> LDAPAccessStorage::mapExternalRolesNoLock(const LDAPClient::Sea
 {
     std::set<String> role_names;
 
+    // If this node can't access LDAP server (or has not privileges to fetch roles) and gets empty list of external roles
+    if (external_roles.size() == 0)
+        return role_names;
+
     if (external_roles.size() != role_search_params.size())
         throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unable to map external roles");
```

Comment:
What if it gets a list of external roles from the initiator, but the configuration file is not the same here so `role_search_params` is not the same here as it was on the initiator and we fail here? I mean maybe this check and also the check for role name's prefix is useful here in general but not for external roles from the initiator?